docs/design/queue-customizable-plugins.md

```diff
@@ -0,0 +1,35 @@
+# Customize Scheduling Algorithms (Plugins) per Queue
+
+## Motivation
+
+The primary use scenario is that different users may need different scheduling algorithms, even in the same clusters. 
+The ideal scenario would be the user being able to specify different actions and plugins per Queue. 
+However, that will introduce considerable changes in the core logic and might not be backward compatible. 
+This project will consist of customizable plugins per Queue as the first step towards that goal.
+Actions will be the same throughout the cluster. However, users can specify different plugins for different Queues. 
+The proposed feature will be a very beneficial feature for Volcano users.
+
+Reference Issue : [Issue 1035](https://github.com/volcano-sh/volcano/issues/1035)
+
+## Solution
+
+* As of now, users can specify actions and plugins through config-map, providing the same scheduling algorithms across the cluster. 
+The project aims to provide an option for users to specify different plugins for different queues. 
+Actions will remain the same as the default configuration, but the user can customize different plugins for different queues. 
+* Currently, the scheduling configuration is loaded initially, and it's only updated when the underlying scheduling config is changed. 
+In a scheduling session, the scheduler traverses through actions ( which were loaded from the scheduling config in the beginning ). 
+In each action, it goes through all jobs ( from all Queues ) that are yet to be scheduled and perform actions. 
+The proposed feature won't affect this workflow. However, which plugins are executed for jobs belonging to a particular Queue will depend on whether any plugins are specified in the Queue YAML or not.
+* If plugins are specified with a Queue, they will be loaded and invoked. Otherwise, default plugins specified in the config-map will be used.
+
+## Implementation
+
+1. Introduce an optional field to specify plugins within Queue CRD spec
```

Comment:
please describe the specific API change in design doc :）

Reply:
is there any update on config file? it's better to define some "type" of scheduling plugins, and refer one of them in Queue spec.

Reply:
We don't need to update the existing config workflow. 
Since we already have `Tier` & `PluginOption` defined. 
We could add the following field to Queue Spec 
```
// optional field to define plugins
Tiers []Tier `yaml:"tiers"`
```

We could also have a `SchedulerConfiguration` instance instead of `Tier`. But we don't need `Actions`, just the plugins.